lib/SIL/InstructionUtils.cpp

```diff
@@ -246,9 +246,52 @@ void ConformanceCollector::scanType(Type type) {
         auto substs = SubType->gatherAllSubstitutions(M.getSwiftModule(),
                                                      nullptr);
         scanSubsts(substs);
+
+        // Check the types of all stored properties of the nominal type.
+        for (VarDecl *Field : NT->getStoredProperties()) {
+          if (Field->hasClangNode() || !Field->hasInterfaceType())
+            continue;
+          CanType FieldTy = SubType->getTypeOfMember(M.getSwiftModule(),
+                                          Field, nullptr)->getCanonicalType();
+          if (needsMetaData(FieldTy))
+            scanType(FieldTy);
+        }
+      }
+    }
+  });
+}
+
+bool ConformanceCollector::needsMetaData(Type type) {
+  type = type->getCanonicalType();
+
+  auto Iter = TypeNeedsMetaData.find(type.getPointer());
+  if (Iter != TypeNeedsMetaData.end())
+    return Iter->second;
+
+  bool &NeedMD = TypeNeedsMetaData[type.getPointer()];
+  NeedMD = false;
+
+  NeedMD = type.findIf([this](Type SubType) ->bool {
+    if (SubType->hasArchetype())
+      return true;
+
+    if (NominalTypeDecl *NT = SubType->getNominalOrBoundGenericNominal()) {
+      if (!NT->hasFixedLayout(M.getSwiftModule(), ResilienceExpansion::Maximal))
+        return true;
+
+      // Check the types of all stored properties of the nominal type.
+      for (VarDecl *Field : NT->getStoredProperties()) {
+        if (Field->hasClangNode() || !Field->hasInterfaceType())
```

Comment:
Can you dump the property and see? This isn't supposed to happen, we might be missing something if the property is not type checked.